tests/unit/modules/test_zypperpkg.py

```diff
@@ -450,6 +452,117 @@ def test_latest_version(self):
+                with patch('salt.modules.zypperpkg.list_pkgs', MagicMock(side_effect=[{"vim": "1.1"}, {"vim": "1.2"}])):
+                    ret = zypper.upgrade(dist_upgrade=True)
+                    self.assertDictEqual(ret, {"vim": {"old": "1.1", "new": "1.2"}})
+                    zypper_mock.assert_any_call(
+                        "dist-upgrade",
+                        "--auto-agree-with-licenses",
+                    )
+
+
+    def test_upgrade_with_novendorchange(self):
+        '''
+        Dist-upgrade without vendor change option.
+        '''
+        with patch.dict(zypper.__grains__, {'osrelease_info': [12, 1]}), \
+                patch('salt.modules.zypperpkg.refresh_db', MagicMock(return_value=True)), \
+                patch('salt.modules.zypperpkg._systemd_scope', MagicMock(return_value=False)):
+            with patch('salt.modules.zypperpkg.__zypper__.noraise.call', MagicMock()) as zypper_mock:
+                with patch('salt.modules.zypperpkg.list_pkgs', MagicMock(side_effect=[{"vim": "1.1"}, {"vim": "1.2"}])):
+                    ret = zypper.upgrade(dist_upgrade=True, novendorchange=True)
+                    zypper_mock.assert_any_call(
+                        "dist-upgrade",
+                        "--auto-agree-with-licenses",
+                    )
+
+    def test_upgrade_with_vendorchange(self):
+        '''
+        Perform dist-upgrade with novendorchange set to False.
+        '''
+        with patch.dict(zypper.__grains__, {'osrelease_info': [12, 1]}), \
+                patch('salt.modules.zypperpkg.refresh_db', MagicMock(return_value=True)), \
+                patch('salt.modules.zypperpkg._systemd_scope', MagicMock(return_value=False)):
+            with patch('salt.modules.zypperpkg.__zypper__.noraise.call', MagicMock()) as zypper_mock:
+                with patch('salt.modules.zypperpkg.__zypper__.allow_vendor_change', MagicMock()) as vendor_change_mock:
+                    with patch(
+                        "salt.modules.zypperpkg.list_pkgs",
+                        MagicMock(side_effect=[{"vim": "1.1"}, {"vim": "1.1"}])
+                    ):
+                        with patch.dict(zypper.__salt__,
+                                        {'pkg_resource.version': MagicMock(return_value='1.15'),
+                                        'lowpkg.version_cmp': MagicMock(return_value=1)}):
+                            ret = zypper.upgrade(
+                                dist_upgrade=True,
+                                dryrun=True,
+                                fromrepo=["Dummy", "Dummy2"],
+                                novendorchange=False,
+                            )
+                            # Since the "call" method is completely mocked away, we need to
+                            # assert on the vendor_change_mock, because the logic of the vendor-change
+                            # flag insertion is in thes __call method.
```

Comment:
`call` is replaced by `zypper_mock`, I think we should check that it is called correctly. `vendor_change_mock` is an implementation detail, I don't think checking is it important. 

I think the goal of this test is to verify that a `zypper.upgrade(xzy, nodeverchange=False)` results in a specific `zypper dist-upgrade` call, isn't it?

Reply:
The `vendor_change_mock` does not exist anymore, rendering this comment outdated.